src/shared_utilities/FileReader.jl

```diff
@@ -1,549 +1,100 @@
-data packaged into a single `PrescribedDataTemporal` struct.
-
-# Arguments
-- `regrid_dirpath`   # directory the data file is stored in.
-- `get_infile`       # function returning path to NCDataset file containing data to regrid.
-- `varnames`         # names of the variables to be regridded.
-- `date_ref`         # reference date to coordinate start of the simulation
-- `t_start`          # start time of the simulation relative to `date_ref` (date_start = date_ref + t_start)
-- `surface_space`    # the space to which we are mapping.
-- `mono`             # flag for monotone remapping of `infile_path`.
-
-# Returns
-- `PrescribedDataTemporal` object
+A struct to efficiently read and process NetCDF files.
 """
-function PrescribedDataTemporal{FT}(
-    regrid_dirpath::String,
-    get_infile::Function,
-    varnames::Vector{String},
-    date_ref::Union{DateTime, DateTimeNoLeap},
-    t_start,
-    surface_space::Spaces.AbstractSpace;
-    mono::Bool = true,
-) where {FT <: AbstractFloat}
-    comms_ctx = ClimaComms.context(surface_space)
-    outfile_root = "temporal_data_cgll"
-
-    # Initialize dummy date to be overwritten by actual dates during file read
-    all_dates = [DateTime(0)]
-
-    # Regrid data at all times from lat/lon (RLL) to simulation grid (CGLL)
-    # Download `infile_path` artifact on root process first to avoid race condition
-    if ClimaComms.iamroot(comms_ctx)
-        infile_path = get_infile()
-        all_dates = Regridder.hdwrite_regridfile_rll_to_cgll(
-            FT,
-            regrid_dirpath,
-            infile_path,
-            varnames,
-            surface_space,
-            outfile_root;
-            mono = mono,
-        )
-
-        NCDataset(infile_path, "r") do ds
-            if !("time" in keys(ds))
-                error(
-                    "Using a temporal albedo map requires data with time dimension.",
-                )
-            end
-        end
-    end
-    all_dates = ClimaComms.bcast(comms_ctx, all_dates)
-    ClimaComms.barrier(comms_ctx)
-    infile_path = get_infile()
-
-    # Init time tracking info
-    data_fields =
-        Fields.zeros(FT, surface_space), Fields.zeros(FT, surface_space)
-    # Store `segment_length` as an array so we can modify it as a field of a struct
-    segment_length = Int[0]
-
-    date_start = date_ref + Dates.Second(t_start)
-    if date_start < all_dates[1]
-        @warn "Simulation start date is before first file data"
-    end
+function NCFileReader(file_path::AbstractString, varname::AbstractString)
+    dataset = NCDatasets.NCDataset(file_path)
 
-    # Find the index of the start file closest to this simulation's start date
-    # Like `segment_length`, store in an array so we can modify in struct
-    date_idx0 =
-        [argmin(abs.(Dates.value(date_start) .- Dates.value.(all_dates[:])))]
+    # NOTE: We are hardcoding a few things here!
+    dimensions = dataset["lon"][:], dataset["lat"][:]
```

Comment:
is this not what `dimensions::DIMS` is for? if not, they could also pass in the dimension_names. then we dont need to hardcode